docs/development/terraform-versions.md

```diff
@@ -0,0 +1,36 @@
+# Terraform Version Compatibility
+
+We use CI to automate testing against *latest* version of each minor version of Terraform
+(eg 1.3.x, 1.4.x, etc that we support).
+
+Given that Hashicorp has changed Terraform licensing to ELv2; and that Linux foundation has forked
+Terraform from 1.5.x to [create OpenTofu](https://opentofu.org/blog/the-opentofu-fork-is-now-available/),
+we are targeting compatibility with 1.6.x feature set.
+
+Do NOT use any features from 1.7.x or later, as we wish to allow all our terraform modules/examples
```

Comment:
Not clear this point. features listed below are introduced in >= 1.4. So not sure if want to say something like 

```suggestion
Do NOT use any features from 1.3.x or later, as we wish to allow all our terraform modules/examples
```

or just clarify why something introduced before 1.7.x (like `strcontains`) cannot be used